src/coprocessor/dag/expr/ctx.rs

```diff
@@ -245,25 +259,63 @@ impl EvalWarnings {
     }
 }
 
+#[derive(Debug, Copy, Clone)]
+pub struct AesModeAttr {
+    pub mode_name: &'static str,
+    pub key_size: usize,
+    pub iv_required: bool,
+}
+
+#[derive(Debug)]
+pub struct AesModes(HashMap<&'static str, AesModeAttr>);
+impl AesModes {
+    pub fn get_mode_attr(&self, aes_name: &str) -> Option<AesModeAttr> {
+        Some(self.0[aes_name])
+    }
+}
+
+const DATA: &'static [(&'static str, &'static str, usize, bool)] = &[
+    ("aes-128-ecb", "ecb", 16, false),
+    ("aes-192-ecb", "ecb", 24, false),
+    ("aes-256-ecb", "ecb", 32, false),
+    ("aes-128-cbc", "cbc", 16, true),
+    ("aes-192-cbc", "cbc", 24, true),
+    ("aes-256-cbc", "cbc", 32, true),
+];
+
 #[derive(Debug)]
 /// Some global variables needed in an evaluation.
 pub struct EvalContext {
     pub cfg: Arc<EvalConfig>,
     pub warnings: EvalWarnings,
+    pub aes_mode: Option<AesModeAttr>,
 }
 
 impl Default for EvalContext {
     fn default() -> EvalContext {
         let cfg = Arc::new(EvalConfig::default());
         let warnings = cfg.new_eval_warnings();
-        EvalContext { cfg, warnings }
+        EvalContext {
+            cfg,
+            warnings,
+            aes_mode: None,
+        }
     }
 }
 
 impl EvalContext {
     pub fn new(cfg: Arc<EvalConfig>) -> EvalContext {
         let warnings = cfg.new_eval_warnings();
-        EvalContext { cfg, warnings }
+        EvalContext {
+            cfg,
+            warnings,
+            aes_mode: None,
+        }
+    }
+
+    pub fn set_aes_mode(&mut self, mode: AesModeAttr) -> &mut Self {
```

Comment:
Seems that I didn't see other places calling this function.

Reply:
Set_aes_mode will be used in the unittest, the last commit code is directly assigned, has been modified.

Reply:
Thanks for explanation. Is this a dynamic variable that will be passed from TiDB later?